extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java

```diff
@@ -826,11 +1091,25 @@ protected RepositoryFile createUniqueFile( final String domainId, final String l
       metadataMap.put( PROPERTY_NAME_LOCALE, locale );
     }
 
+    addDataSourceType( metadataMap, data );
+
+    // Create the new file
+    final RepositoryFile file = repository.createFile( getMetadataDir().getId(),
+            new RepositoryFile.Builder( filename ).build(), data, null );
+
     // Update the metadata
     repository.setFileMetadata( file.getId(), metadataMap );
     return file;
   }
 
+  public RepositoryFile updateFile( RepositoryFile domainFile, SimpleRepositoryFileData data ) {
```

Comment:
[#3 creating a new Domain object #createUniqueFile and #storeDomain](https://jira.pentaho.com/browse/BACKLOG-36129)